### ISSUE TITLE
Try to use packit to generate copr builds

### ISSUE THREAD

**Issue by mulkieran** [stratis-storage/project#309]
This is a little challenging compared to some other moves we've done in the recent past. But it makes sense to think of the blackbox tests as really two tasks: building the rpm and then unpacking the rpm and actually running the tests.

Does it make sense to build the rpm as part of a nightly GitHub Action? Can we then post it in a well-known location so another task can use it? Building as part of a GitHub Action seems eminently possible, posting requires permissions.

Running the tests is a different question. The tests try to mimic exactly behavior on an installed system, and that includes starting and stopping the individual tests via systemd. systemd is not available on the docker containers that we have been accustomed to use but is supported via Podman and there is some Podman support in GitHub Actions now: https://github.com/actions/virtual-environments/issues/320. So running the tests might be possible, so long as the container could be persuaded to present enough storage devices.

**Comment by mulkieran** [comment]
We think that a nightly task that generated an rpm as an artifact could then be run on copr, and that would test our packaging well.

**Comment by mulkieran** [comment]
The next step is to pull the spec file from the Fedora rawhide branch; then update it so it bumps the release, then build the rpm. Also, a bunch of the finding what the most recent tag is and then checking out from that tag that exists in the workflow can go away, as the point in nighly will be just to generate the rpm from the most recently committed version. I bet there exists a tool to bump the release version in a fedora spec file.

**Comment by mulkieran** [comment]
rpmdev-bumpspec

**Comment by mulkieran** [comment]
Blocked by #339

**Comment by mulkieran** [comment]
We can possibly consider using packit.

**Comment by mulkieran** [comment]
We should try at least doing copr builds, nightly, via GitHub Actions.

**Comment by lachmanfrantisek** [comment]
Hi @mulkieran, Packit lead here!

Let me know if you need any help with the setup. If you are interested in Copr builds and running tests on top of it, then Packit is most probably the right tool for that.

We are more than happy to help you set everything up. (It's usually a few lines of the config file.) The easiest way to contact us is on our team channel `#packit:fedora.im` on Matrix.

The following guide should help you with the basic onboarding: https://packit.dev/docs/guide/

**Comment by mulkieran** [comment]
I think we have succeeded.
